xarray/core/dataarray.py

```diff
@@ -1399,6 +1399,66 @@ def unstack(self, dim=None):
         ds = self._to_temp_dataset().unstack(dim)
         return self._from_temp_dataset(ds)
 
+    def to_unstacked_dataset(self, dim, level=0,
+                             variable_dim='variable'):
+        """Unstack DataArray expanding to Dataset along a given level of a
+        stacked coordinate.
+
+        This is the inverse operation of Dataset.to_stacked_array.
+
+        Parameters
+        ----------
+        dim : str
+            Name of existing dimension to unstack
+        level : int
```

Comment:
As we all love labels, it would be nice if `level` could also accept level names (`str`) :-). Replacing``idx.levels[level]`` with ``idx.get_level_values(level)`` below would be enough, I think.

Reply:
I agree it would be nice to do this. I would prefer not to overload the `level` argument, so maybe we a `label` argument could take precedence if given.

Reply:
It seems like `idx.get_level_values` has a potential problem. What happens if the labels are also integers?

Reply:
Fine...if it's good enough for pandas, it's good enough for us I guess. Updated the docs and use `idx.get_level_values(level)` as you sugget.